### PR TITLE
Add Custom Asset to ITIL Link Types from Definition

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Selection_362

### CODE REVIEW

File: src/Glpi/Asset/AssetDefinition.php [hunk 1]
```diff
@@ -621,6 +626,43 @@ private function syncProfilesRights(): void
                 ProfileRight::updateProfileRights($profile_id, [$rightname => $rights]);
             }
         }
+
+        if (isset($this->input['profiles_extra'])) {
```
[the remark this answers is not in the document]
Comment (cconard96) [reply src/Glpi/Asset/AssetDefinition.php]: I added the early return and simplified the handling of the extra fields a bit.

**Review by AdrienClairembault — APPROVED** [review]

**Comment by orthagh** [comment]
The design seems a bit odd to me, I said the same thing yesterday in Caen office (I forgot to write my opinion since)
I'm not a big fan of the "Show extra fields" toggle that seems to be here to hide the mess under :p

Some suggestions:
 - maybe we could find a shorter label and try to include it in the main rows
 - otherwise, maybe align the thing to the left, and remove the border-bottom of the above rows

In both case, the top toggle seems not usefull

**Comment by cconard96** [comment]
>     * maybe we could find a shorter label and try to include it in the main rows

I tried that first. It seemed messy to handle the display of the checkbox matrix and handling the permissions on the server side. It also means that we have a lot of horizontal scrolling. Finally, I don't know how many options like this we will end up adding (or have plugins add), but maybe not all can be displayed as a checkbox.

>     * otherwise, maybe align the thing to the left, and remove the border-bottom of the above rows

The alignment wasn't intentional, but it does help show that it is part of the Profile listed above it. Removing the border is a good idea. It would require some changes to the checkbox matrix template to allow it.

> 
> In both cases, the top toggle seems not useful

It was just a way to hide the extra fields if the interest is only in the basic permissions. If you have a lot of Profiles and/or if we add a lot more of these extra fields, it could be a lot of scrolling to find the Profiles you want without using the browser search. Maybe a search box to filter the displayed Profiles would be better?

**Comment by cedric-anne** [comment]
And what about displaying a unique field at the end of the rights matrix that would permit to select the list of profiles that should be able to use this kind of asset in ITIL context ?

**Comment by orthagh** [comment]
maybe an idea yes. Note sure atm, but I take other opinions

**Comment by cedric-anne** [comment]
ping @orthagh 

What should we do with this PR?

**Comment by orthagh** [comment]
After discussing a bit with @cedric-anne, this is due to the fact we enabled, long time ago, an option on github to rebase the pr before launching the tests suite.

The pr itself indeed doesn't fail locally, but after a rebase, yes.

I don't have a strong opinion about the option, but you can get the fail and fix it after a rebase.
I'm not sure if we can keep the option as you must know the thing, and even after insisting to everyone about that, I think there still will be cases where we forget about it.

**Comment by orthagh** [comment]
Here is the popularity of my request :p

So let's go for a unified multiple dropdown where you select profiles that can assign items.
It will take reduced space and it should be a better experience.

**Comment by orthagh** [comment]
So for this one, apart from fixing tests, We may review the label a bit. I suggest:
"Profiles that can associate %s with tickets, problems or changes", where %s is the asset name/label

**Review by orthagh — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
I made some small changes.

1. The field is configurable even when the asset definition is not active. It does not cause any issue, and it could permit to fully configure the definition before activating it. It also permit to remove the `Some fields may not be available because the asset definition is inactive.` message that was not relevant for the custom dropdown case.

2. I fixed the `Self-service` case that was not visible in the selection because it was not retrieved in the initial DB request (filtered on `interface != helpdesk` ).

3. I used a `_` prefix for the `profiles_extra` input.

4. I used `Profile::update()` to processto the update, in order to add an entry in the profile history.

**Pull request merged**